util/EIP-2124/src/lib.rs

```diff
@@ -216,113 +266,155 @@ mod tests {
+		filter.set_head(7_987_396);
+		assert_eq!(filter.is_compatible(ForkId { hash: ForkHash(0xafec_6b27), next: 0 }), Err(RejectReason::LocalIncompatibleOrStale));
 
 		// Local is mainnet Petersburg, far in the future. Remote announces Gopherium (non existing fork)
 		// at some future block 88888888, for itself, but past block for local. Local is incompatible.
 		//
 		// This case detects non-upgraded nodes with majority hash power (typical Ropsten mess).
-		let mut far_away_petersburg = petersburg_filter.clone();
-		far_away_petersburg.head = 88888888;
-		assert_eq!(far_away_petersburg.is_valid(ForkId { hash: ForkHash(0x668db0af), next: 88888888 }), Err(RejectReason::LocalIncompatibleOrStale));
+		filter.set_head(88_888_888);
+		assert_eq!(filter.is_compatible(ForkId { hash: ForkHash(0x668d_b0af), next: 88_888_888 }), Err(RejectReason::LocalIncompatibleOrStale));
 
 		// Local is mainnet Byzantium. Remote is also in Byzantium, but announces Gopherium (non existing
 		// fork) at block 7279999, before Petersburg. Local is incompatible.
-		assert_eq!(byzantium_filter.is_valid(ForkId { hash: ForkHash(0xa00bc324), next: 7279999 }), Err(RejectReason::LocalIncompatibleOrStale));
+		filter.set_head(7_279_999);
+		assert_eq!(filter.is_compatible(ForkId { hash: ForkHash(0xa00b_c324), next: 7_279_999 }), Err(RejectReason::LocalIncompatibleOrStale));
 	}
 
 	#[test]
-	fn test_forkid_serialization() {
+	fn forkid_serialization() {
 		assert_eq!(rlp::encode(&ForkId { hash: ForkHash(0), next: 0 }), hex!("c6840000000080"));
-		assert_eq!(rlp::encode(&ForkId { hash: ForkHash(0xdeadbeef), next: 0xBADDCAFE }), hex!("ca84deadbeef84baddcafe"));
+		assert_eq!(rlp::encode(&ForkId { hash: ForkHash(0xdead_beef), next: 0xBADD_CAFE }), hex!("ca84deadbeef84baddcafe"));
 		assert_eq!(rlp::encode(&ForkId { hash: ForkHash(u32::max_value()), next: u64::max_value() }), hex!("ce84ffffffff88ffffffffffffffff"));
 
 		assert_eq!(rlp::decode::<ForkId>(&hex!("c6840000000080")).unwrap(), ForkId { hash: ForkHash(0), next: 0 });
-		assert_eq!(rlp::decode::<ForkId>(&hex!("ca84deadbeef84baddcafe")).unwrap(), ForkId { hash: ForkHash(0xdeadbeef), next: 0xBADDCAFE });
+		assert_eq!(rlp::decode::<ForkId>(&hex!("ca84deadbeef84baddcafe")).unwrap(), ForkId { hash: ForkHash(0xdead_beef), next: 0xBADD_CAFE });
 		assert_eq!(rlp::decode::<ForkId>(&hex!("ce84ffffffff88ffffffffffffffff")).unwrap(), ForkId { hash: ForkHash(u32::max_value()), next: u64::max_value() });
 	}
+
+	#[test]
+	fn compute_cache() {
+		let b1 = 1_150_000;
+		let b2 = 1_920_000;
+
+		let h0 = ForkId { hash: ForkHash(0xfc64_ec04), next: b1 };
+		let h1 = ForkId { hash: ForkHash(0x97c2_c34c), next: b2 };
+		let h2 = ForkId { hash: ForkHash(0x91d1_f948), next: 0 };
+
+		let mut fork_filter = ForkFilter::new(
+			0,
+			GENESIS_HASH,
+			vec![b1, b2],
+		);
+
+		assert!(!fork_filter.set_head_priv(0));
```

Comment:
why are we using `set_head_priv` here instead of `set_head`?

Reply:
`set_head_priv` returns boolean that indicates whether fork filter transitioned into another epoch and the cache was recomputed. This is an internal implementation detail that I only exposed for the testing purposes in this private method. End users should only care about `set_head`.